src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -46,5 +46,22 @@ CORECLR_HOSTING_API(coreclr_execute_assembly,
             unsigned int* exitCode);
 
 #undef CORECLR_HOSTING_API
-                      
+
+struct SequencePointInfo
+{
+    int lineNumber, ilOffset;
+    char16_t* fileName;
+};
+
+struct MethodDebugInfo
+{
+    SequencePointInfo* points;
+    int size;
+};
+
+typedef int (*GetInfoForMethodDelegate)(const char*, unsigned int, MethodDebugInfo& methodDebugInfo);
+extern GetInfoForMethodDelegate getInfoForMethodDelegate;
```

Comment:
We should add an API to set the delegate instead of having the variable in the hosting API.